### DISCUSSION TITLE
More flexible data version management utilies, such as use db to store all commited dvc

### DISCUSSION THREAD

**Discussion by NerdToMars** [treeverse/dvc#6730] [New Features & Ideas]
For some data centrical applications, after we commit a lot of different versions of data into git, a bit inconvenient for engineers or administrators to visualize some features of history data.  Maybe the following ideas will help us solve this pain point:
1. when dvc commits or adds, engineers can add more metadata of this version, such as data schema or distribution. 
>  more general ideas such as allowing engineers to provide a call back to generate .dvc files

2. In python API, developers can retrieve all metadata of each version. I am using the git tag to get a list of version tags, and then I can use dvc read to iterate metadata in all version tags. ( not sure if there is any better solution)
3. maybe we can add dvc info into a DB or allow users to connect to a DB?

**Comment by efiop** [comment]
Converting to discussion https://github.com/iterative/dvc/discussions/6730
